tests/lib/EventDispatcher/EventListener/ContentSuggestionSubscriberTest.php

```diff
@@ -8,84 +8,230 @@
+        $this->capturedQuery = null;
+        $this->searchServiceSupportsScoring = false;
+        $this->loggerMock = $this->createMock(LoggerInterface::class);
+    }
+
     public function testSubscribedEvents(): void
     {
-        $this->assertSame(
-            [BuildSuggestionCollectionEvent::class => 'onBuildSuggestionCollectionEvent'],
+        self::assertSame(
+            [
+                BuildSuggestionCollectionEvent::class => 'onBuildSuggestionCollectionEvent',
+            ],
             ContentSuggestionSubscriber::getSubscribedEvents()
         );
     }
 
-    public function testOnContentSuggestion(): void
+    public function testOnContentSuggestionWithLegacyEngineAndScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = true;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithNonLegacyEngineAndScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'not_legacy']]);
+        $this->searchServiceSupportsScoring = true;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithLegacyEngineWithoutScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = false;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithNonLegacyEngineWithoutScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'not_legacy']]);
+        $this->searchServiceSupportsScoring = false;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithException(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = true;
+
+        $query = new SuggestionQuery('test', 10, 'eng-GB');
+
+        $searchService = $this->getSearchServiceMockWithException();
+        $mapper = $this->getSearchHitToContentSuggestionMapperMock();
+        $sortingDefinitionRegistry = $this->getSortingDefinitionRegistryMock();
+
+        $subscriber = new ContentSuggestionSubscriber($searchService, $mapper, $sortingDefinitionRegistry);
+        $subscriber->setLogger($this->loggerMock);
+
+        $event = new BuildSuggestionCollectionEvent($query);
+
+        $this->loggerMock
+            ->expects($this->once())
+            ->method('error');
```

Comment:
What will be the message?

```suggestion
        $this->loggerMock
            ->expects(self::once())
            ->method('error')
            ->with(...);
```

Will the exception, if any, be logged into it?